src/disc/BlockedDiscretization.hpp

```diff
@@ -366,25 +377,165 @@ class BlockedDiscretization : public AbstractDiscretization
   Teuchos::RCP<Thyra_MultiVector>
   getBlockedSolutionMV(const bool overlapped = false) const;
 
-  void getField (Thyra_Vector& field_vector, const std::string& field_name) const {
-    return m_blocks[0]->getField(field_vector, field_name);
+// GAH - These all need a serious look, they are just stubbed in to get things compiling
+
+  //! Retrieve coordinate vector (num_used_nodes * 3)
+  const Teuchos::ArrayRCP<double>&
+  getCoordinates() const {
+     return m_blocks[0]->getCoordinates();
+  }
+
+  Teuchos::RCP<Thyra_Vector>
+  getSolutionField(bool overlapped) const
+  {
+    return m_blocks[0]->getSolutionField(overlapped);
   }
 
-  void setField(
-      const Thyra_Vector& field_vector,
-      const std::string&  field_name,
-      const bool          overlapped = false) {
-    m_blocks[0]->setField(field_vector, field_name, overlapped);
+  Teuchos::RCP<Thyra_MultiVector>
+  getSolutionMV(bool overlapped) const
+  {
+    return m_blocks[0]->getSolutionMV(overlapped);
   }
 
-  //! used when NetCDF output on a latitude-longitude grid is requested.
-  // Each struct contains a latitude/longitude index and it's parametric
-  // coordinates in an element.
-  struct interp
+  void
+  getField(Thyra_Vector& result, const std::string& name) const
   {
-    std::pair<double, double>     parametric_coords;
-    std::pair<unsigned, unsigned> latitude_longitude;
-  };
+    m_blocks[0]->getField(result, name);
+  }
+
+  void
+  getSolutionField(Thyra_Vector& result, const bool overlapped) const
+  {
+    m_blocks[0]->getSolutionField(result, overlapped);
```

Comment:
These are ok now that we have a single block. Later on we should:
1. check that the input vector is a Thyra::ProductVector
2. check that the input blk vector and m_blocks have the same number of blocks
3. loop over all blocks, and get call the corresponding method in m_blocks[i] passing only the i-th block of the input vector.

Reply:
Agreed,